common/k8stest/util_daemonset.go

```diff
@@ -146,6 +151,42 @@ func WaitForDaemonsetReady(dsName string, namespace string, sleepTime int, durat
 	return ready
 }
 
+func DeleteDaemonsetAndWaitPodDeletion(dsName string, namespace string, sleepTime int, duration time.Duration) bool {
+
+	ds, err := GetDaemonSet(dsName, namespace)
+	if err != nil {
+		logf.Log.Info("Failed to get daemonset", "daemonset", dsName, "namespace", namespace, "error", err)
+		return false
+	}
+
+	dsPodList, err := ListPodsByPrefix(ds.Namespace, ds.Name)
+	if err != nil {
+		logf.Log.Info("Failed to list pods with daemonset prefix", "daemonset", dsName, "error", err)
+		return false
+	}
+
+	err = DeleteDaemonSet(dsName, namespace)
```

Comment:
Should the pod list be done after deleting the daemonset?

Reply:
Pod list corresponding to daemonset is done before deletion of daemonset just to verify that previous daemonset pods does not exist after daemonset deletion